code-conventions.md

```diff
@@ -163,33 +194,79 @@ Your comments should look something like these:
 ## Language Features
 
 ### Compatibility
-* Contributions should maintain Java 8 source/binary compatibility, even though compiling Skript requires Java 21
-  - Users must not need JRE newer than version 8
+[//]: # (To be updated after feature/2.9 for Java 17)
+* Contributions should maintain Java 11 source/binary compatibility, even though compiling Skript requires Java 21
+  - Users must not need JRE newer than version 11
 * Versions up to and including Java 21 should work too
   - Please avoid using unsafe reflection
 * It is recommended to make fields final, if they are effectively final
-* Local variables and method parameters should not be declared final
+* Local variables and method parameters should not be declared final unless used in anonymous classes, lambdas
+or try-with-resources sections where their immutability is necessary
 * Methods should be declared final only where necessary
 * Use `@Override` whenever applicable
-
-### Nullness
+  - `@Override` annotations may be omitted to prevent compilation errors when something overrides only
+  on a version-dependent basis (e.g. in Library XYZ version 2 we override `getX()` but in version 3 it's
+  gone, and we call it ourselves)
+
+### null-ness
+* We use **JetBrains** Annotations for specifying null-ness and method contracts.
+  * If editing a file using a different annotation set (e.g. Javax, Eclipse Sisu, Bukkit)
+  these should be replaced with their JetBrains equivalent.
+  * The semantics for JetBrains Annotations are strict _and should be observed!_
+    * Many IDEs have built-in compiler-level support for these, and can even be set to produce strict
+    errors when an annotation is misused; do not misuse them.
+  * **`@NotNull`**
+    * > An element annotated with NotNull claims null value is forbidden to return (for methods), 
+    pass to (parameters) and hold (local variables and fields).
+	* Something is `@NotNull` iff it is never null from its inception (new X) to its garbage collection, 
```

Comment:
```suggestion
	* Something is `@NotNull` if it is never null from its inception (new X) to its garbage collection, 
```

Reply:
<img width="327" alt="image" src="https://github.com/SkriptLang/Skript/assets/14147477/00c5bf8d-ae78-485d-a673-2ed8619bd626">

Reply:
Ah, I never knew about this